docs/developers/serializers.rst

```diff
@@ -0,0 +1,187 @@
+the result differently, then a matching serializer will likely be required.
+
+.. code-block:: python
+
+   from pootle_store.store.serialize import StoreSerializer
+
+
+   class CustomSerializer(StoreSerializer):
+
+       @property
+       def output(self):
+           if self.store.id % 2:
+               return " %s " % self.original_data
+           return self.original_data
+
+
+.. _serializers#apps:
+
+Adding an application configuration
+-----------------------------------
+
+If you do not already have an `apps` configuration for your application you will need
+to add one.
+
+For an application named `pootle_custom` you need to add lines similar to the following
+to your `__init__.py`:
+
+
+.. code-block:: python
+
+   default_app_config = 'pootle_custom.apps.PootleCustomConfig'
+
+In a file named `apps.py` you will need to add the django configuration for your
+application.
+
+To add a serializer you will need to provide some pootle `providers`, A configuration
+such as the following will activate the necessary code.
+
+.. code-block:: python
+
+   import importlib
+
+   from django.apps import AppConfig
+
+
+   class PootleCustomConfig(AppConfig):
+
+       name = "pootle_custom"
+       verbose_name = "Pootle Custom"
+
+       def ready(self):
+           importlib.import_module("pootle_custom.providers")
+
+
+.. _serializers#serializer-providers:
+
+Add a provider file to enable your serializers
+----------------------------------------------
+
+To enable serializers you will need to add a provider function
+for `pootle.core.delegate.serializers`.
+
+Add a file called `providers.py` with the following code
+
+Serializers are named, and the provider function should return
+a dictionary of serializer classes, using the serializer names
+as the dictionary keys.
+
+
+.. code-block:: python
+
+   from pootle.core.delegate import serializers
+   from pootle.core.plugin import provider
+
+   from pootle_custom.serializers import CustomSerializer
+
+   
+   @provider(serializers, sender=Store)
+   def provide_serializers(**kwargs):
+       return dict(custom_serializer=CustomSerializer)
+
+
+.. _serializers#deserializer-providers:
+
+Enabling your deserializers
+---------------------------
+
+To enable deserializers you will need to add a provider function
+for `pootle.core.delegate.deserializers`.
+
+Deserializers are named, and the provider function should return
+a dictionary of deserializer classes, using the deserializer names
+as the dictionary keys.
+
+If necessary, add a file called `providers.py`. To enable our CustomSerializer
+we will need something like the following in the providers file
+
+
+.. code-block:: python
+
+   from pootle.core.delegate import serializers
+   from pootle.core.plugin import provider
+
+   from pootle_custom.serializers import CustomDeserializer
+
+   
+   @provider(serializers, sender=Store)
+   def provide_serializers(**kwargs):
```

Comment:
these are not really functional bits of code, they are examples, im really not minded to put them in standalone scripts